physlr/physlr.py

```diff
@@ -1674,62 +1800,129 @@ def physlr_molecules(self):
                 int(timeit.default_timer() - t0),
                 "Separating junction-causing barcodes into molecules "
                 "using the following algorithm(s):\n\t",
-                self.args.strategy.replace("+", " + "),
+                alg_list_2d,
                 "\n\tand other barcodes with bc.",
                 file=sys.stderr)
         else:
             gin = self.read_graph(self.args.FILES)
             print(
                 int(timeit.default_timer() - t0),
                 "Separating barcodes into molecules using the following algorithm(s):\n\t",
-                self.args.strategy.replace("+", " + "),
+                alg_list_2d,
+                # self.args.strategy.replace("++"," / ").replace("+", " + "),
                 file=sys.stderr)
 
         Physlr.filter_edges(gin, self.args.m)
+        round_num = self.args.round
+        for alg_list in alg_list_2d:
+            if Physlr.args.set_settings and len(alg_list_2d) > 1:
```

Comment:
We could have used `self` since both are defined in this scope. I'll just change it.
Is there any python-specific difference between using the name of the class or `self`? I am asking to learn :D and we also have some other instances like this. lines 1849 to 1851.

Reply:
This is an instance method (specific to this instance of the Physlr class), so it has the access to all of the instance variables (with self). That is in contrast to the static methods, which don't have direct access to data about the class/instance. So in this case, we do have access to the instance variables.

Reply:
yep, right. I'll change this and some other occurences..